src/Clock/ScaleClock.php

```diff
@@ -13,21 +13,11 @@
  */
 final class ScaleClock implements Clock
 {
-    /**
-     * The reference clock.
-     */
-    private readonly Clock $referenceClock;
-
     /**
      * The start time.
      */
     private readonly Instant $startTime;
```

Comment:
While I'm quite a heavy user of promoted properties these days, I'm still a bit annoyed when I see a mix of promoted an non-promoted properties, I feel like this makes the class a bit harder to read. WDYT?

Reply:
I've gotten used to it, and in a sense, I like that it separates "dependencies" of an object (what it needs in order to be created) and "internal properties" (used for logic, shortcuts, etc).

Reply:
I agree with you when used in services. In this particular case, I'm less convinced. It's pretty much the only one that bothers me in the codebase though, I guess I'll have to learn to live with it!